packages/framework/src/Component/Money/Money.php

```diff
@@ -227,6 +227,10 @@ public function isZero(): bool
      */
     protected static function createDecimal($value, ?int $scale = null): Decimal
     {
+        if (is_int($value) == false && is_string($value) == false) {
```

Comment:
You should have strict comparison here

Reply:
Ok, I changet it to Webmoart:assert - e66ed7f15a9d326866c6fca0429852bdd668d6a1